addons/web/static/src/model/relational_model/dynamic_group_list.js

```diff
@@ -261,6 +261,7 @@ export class DynamicGroupList extends DynamicList {
             for (const group of groups) {
                 this._removeGroup(group);
             }
+            this.model._updateConfig(this.config, { groups: configGroups }, { reload: false });
```

Comment:
Do you still need _removeGroup ? 
You try to do the same thing ? without the count--